src/mono/mono/utils/options-def.h

```diff
@@ -118,6 +118,13 @@ DEFINE_BOOL(jiterpreter_eliminate_null_checks, "jiterpreter-eliminate-null-check
 DEFINE_BOOL(jiterpreter_backward_branches_enabled, "jiterpreter-backward-branches-enabled", TRUE, "Enable performing backward branches without exiting traces")
 // Attempt to use WASM v128 opcodes to implement SIMD interpreter opcodes
 DEFINE_BOOL(jiterpreter_enable_simd, "jiterpreter-simd-enabled", TRUE, "Attempt to use WebAssembly SIMD support")
+#if DEBUG
+// In debug builds emscripten puts the stack first for debugging reasons, so this isn't safe
+DEFINE_BOOL_READONLY(jiterpreter_zero_page_optimization, "jiterpreter-zero-page-optimization", FALSE, "Exploit the zero page being unallocated to optimize out null checks")
+#else
+// Since the zero page is unallocated, loading array/string/span lengths from null ptrs will yield zero
+DEFINE_BOOL(jiterpreter_zero_page_optimization, "jiterpreter-zero-page-optimization", TRUE, "Exploit the zero page being unallocated to optimize out null checks")
+#endif
```

Comment:
This... seems wrong. this `DEBUG` ifdef will be set when compiling C to object files, but I think it's probably the optimization of the final link step that will determine if `emcc.py` sets `STACK_FIRST` to true/false.  And that can be influenced by downstream users using `-p:WasmBuildNative=true` (or doing AOT compilation) and possibly messing with `-p:EmccFlags`

Reply:
That's a good point, even if we set STACK_FIRST a downstream user relinking the runtime might not set it. We would need some way to know whether the stack is there reliably.

Reply:
emscripten has a GLOBAL_BASE option to set where static data begins.

Reply:
https://github.com/emscripten-core/emscripten/pull/18057/files

Reply:
Yeah, but someone relinking us downstream might override it. I think maybe we could programmatically detect where GLOBAL_BASE actually is by inspecting our module.

Reply:
https://emscripten.org/docs/api_reference/stack.h.html It looks like we should be able to use this.